### ISSUE TITLE
[RFE] respect `target` option in `mount` type secret

### ISSUE THREAD

**Issue by mskarbek** [containers/podman#12287]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

`podman run -it --secret token,type=env,target=TOKEN ubi8/ubi:latest bash` is currently a valid command.
`podman run -it --secret token,type=mount,target=TOKEN ubi8/ubi:latest bash` ends with error:
```
Error: target option is invalid for mounted secrets: error parsing secret
```
I would like to be able to specify `target` option in `mount` type as file name. The above command should produce file `/run/secrets/TOKEN` in that instance.

Use case:
Multiple containers of the same image, each configured individually with their secrets but with a common internal script/unit file. I have a unit file that always looks in the same place, but each container gets their individual secret.

Edit:
This should also include `mount-env` (#12284) when/if implemented.

**Comment by rhatdan** [comment]
@ashley-cui PTAL

**Comment by rhatdan** [comment]
@flouthoc PTAL

**Comment by rhatdan** [comment]
@mskarbek interested in opening a PR to allow this?

**Comment by mskarbek** [comment]
@rhatdan is there a documentation describing code structure, or maybe recording of a codding session that shows of code structure during implementation of some feature? I have written little to no Go code, but I'll gladly learn something new.

**Comment by flouthoc** [comment]
@mskarbek I think this is a good RFE and needs a small patch. @rhatdan @ashley-cui  PTAL

**Comment by mskarbek** [comment]
Can this be back ported to 3.4.2-rhell branch/included in 3.4 release in RHEL?

**Comment by mheon** [comment]
I recommend you open a BZ if you want a backport.
